x-pack/plugins/canvas/public/components/function_reference_generator/function_reference_generator.tsx

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC } from 'react';
+import { AnyExpressionFunctionDefinition } from 'src/plugins/expressions';
+import { EuiButtonEmpty } from '@elastic/eui';
+// @ts-ignore untyped lib
+import pluralize from 'pluralize';
+import copy from 'copy-to-clipboard';
+import { functions as browserFunctions } from '../../../canvas_plugin_src/functions/browser';
+import { functions as serverFunctions } from '../../../canvas_plugin_src/functions/server';
+import { ArgumentType } from '../../../types';
+import { notifyService } from '../../services';
+import { isValidDataUrl, DATATABLE_COLUMN_TYPES } from '../../../common/lib';
+import { getFunctionExamples, FunctionExample } from './examples';
+
+const ALPHABET = 'abcdefghijklmnopqrstuvwxyz'.split('');
+const REQUIRED_ARG_ANNOTATION = '***';
+const MULTI_ARG_ANNOTATION = '†';
+const UNNAMED_ARG = '_Unnamed_';
+const ANY_TYPE = '`any`';
+
+const examplesDict = getFunctionExamples();
+
+const fnList = [
+  ...browserFunctions.map((fn) => fn().name),
+  ...serverFunctions.map((fn) => fn().name),
+  'asset',
+  'filters',
+  'timelion',
+  'to',
+  'font',
+  'var',
+  'var_set',
+  // ignore unsupported embeddables functions for now
+].filter((fn) => !['savedSearch'].includes(fn));
+
+interface FunctionDictionary {
+  [key: string]: AnyExpressionFunctionDefinition[];
+}
+
+const wrapInBackTicks = (str: string) => `\`${str}\``;
+const wrapInDoubleQuotes = (str: string) => (str.contains('"') ? str : `"${str}"`);
```

Comment:
should this be "contains" or "includes"? I get an exception on this line when I run it